src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/TaskResourceQueryBuilder.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.cft.query;
+
+import uk.gov.hmcts.reform.wataskmanagementapi.cft.entities.TaskResource;
+
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Order;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+
+public abstract class TaskResourceQueryBuilder<T> {
+    protected final EntityManager entityManager;
+    protected final CriteriaBuilder builder;
+    protected final CriteriaQuery<T> query;
+    protected final Root<TaskResource> root;
```

Comment:
This need to be generic type T, isn't it?. When you declare it as TaskResource, then TaskResourceQueryBuilder can only be used with TaskResource.